### PR TITLE
EZP-27902: As an Editor, I want to edit a Date field type with the new UI guidelines

### PR DESCRIPTION
**Pull request by sunpietro**
> JIRA: https://jira.ez.no/browse/EZP-27902
> Requires: https://github.com/ezsystems/repository-forms/pull/170

**THINGS DONE:**
- [x] implemented frontend part,
- [x] implemented backend part.

### CODE REVIEW

**Comment by webhdx** [comment]
Backend part (in ezsystems/repository-forms) is finished: https://github.com/ezsystems/repository-forms/pull/169

**Comment by sunpietro** [comment]
Everything is done. 
@webhdx There's still an issue with storing default _current time_ as the field value and with restoring a correct time after getting back to content edit.

File: src/bundle/Resources/public/js/scripts/fieldType/ezdate.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    const dateConfig = {
+        formatDate: (date) => (new Date(date)).toLocaleDateString()
+    };
+    const updateInputValue = (sourceInput, date) => {
+        sourceInput.value = Math.floor((new Date(date)).getTime() / 1000);
+        sourceInput.dispatchEvent(new CustomEvent(EVENT_VALUE_CHANGED));
+    };
+    const initFlatPickr = (field) => {
+        const sourceInput = field.querySelector(SELECTOR_INPUT);
+
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/fieldType/ezdate.js]: remove empty line

**Comment by barbaragr** [comment]
New ticket for day difference: https://jira.ez.no/browse/EZP-28244

**Pull request merged**